doc

justinmk opened the pull request:
- [x]  lua-concepts
- [x] https://github.com/neovim/neovim/pull/12704
- [x] https://github.com/neovim/neovim/issues/12875
- [x] update nvim-lspconfig URL https://github.com/neovim/neovim/pull/12823#issuecomment-689488981
- [x] https://github.com/neovim/neovim/pull/13972

clason commented:
@justinmk you might be interested in https://github.com/nanotee/nvim-lua-guide

justinmk commented:
> you might be interested in https://github.com/nanotee/nvim-lua-guide

wow that looks good actually. Also a reminder that our docs could be more accessible if their HTML form was improved... 

but that guide really does look like someone put thought into it. nice.

clason commented:
> wow that looks good actually. Also a reminder that our docs could be more accessible if their HTML form was improved...

I do think that would be worth someone (label:good-first-issue) spending a bit of time on -- this would make it much more accessible via Google search, I believe. What would also be helpful is a sidebar with a table of contents and a search bar (like https://docs.julialang.org/en/v1/ or https://fenics.readthedocs.io/en/latest/).

amenat commented:
@justinmk  Inbuilt documentation for LSP points to old url of lsp config repo: https://github.com/neovim/nvim-lsp while github usually redirects renamed repositories it would good idea to update documentation to point to current URL: https://github.com/neovim/nvim-lspconfig

This also tends to be source of confusion about whether this is a plugin that enables LSP or just bunch of configs.

Reference: https://github.com/neovim/neovim/blob/d8e6a0396a1bdc19c9e15adb233540816274487d/runtime/doc/lsp.txt#L28

rockerBOO commented:
#12885 merged to complete task `nvim-lspconfig`.

justinmk committed:
docs: various

- increase python line-length limit from 88 => 100.
- gen_help_html: fix bug in "tag" case (tbl_count => tbl_contains)

ref #15632
fix #18215
fix #18479
fix #20527
fix #20532

Co-authored-by: Ben Weedon <ben@weedon.email>

justinmk commented on runtime/doc/provider.txt:
```diff
@@ -236,6 +236,22 @@ The "copy" function stores a list of lines and the register type. The "paste"
+For Windows WSL, try this g:clipboard definition:
+>
+    let g:clipboard = {
+                \   'name': 'WslClipboard',
```
This is from https://www.reddit.com/r/neovim/comments/g94zrl/solution_neovim_clipboard_with_wsl/ . Can someone using WSL confirm that this works? Or is there a better solution these days?

dundargoc replied:
I tried it out, it works but there is a non-insignificant delay for me personally when pasting. Using win32yank removes most of the delay, so maybe it could be worth mentioning?

```
let g:clipboard = {
            \   'name': 'WslClipboard',
            \   'copy': {
            \      '+': 'win32yank.exe -i --crlf',
            \      '*': 'win32yank.exe -i --crlf',
            \    },
            \   'paste': {
            \      '+': 'win32yank.exe -o --lf',
            \      '*': 'win32yank.exe -o --lf',
            \   },
            \   'cache_enabled': 0,
            \ }
```

 I will leave it up to you what to do with this information :)

dundargoc replied:
Derp, ignore what I said. win32yank is apparently already automatically used if it's installed so the above code snippet doesn't seem to be needed.

zeertzjq commented on runtime/doc/develop.txt:
```diff
@@ -260,37 +262,50 @@ If the function acts on an object then {thing} is the name of that object
+    `nvim_get_keymap('v')` operates in a global context (first parameter is not
+    a Buffer). The "get" {action} indicates that it gets anything matching the
+    given filter parameter. There is no need for a "list" action because
+    `nvim_get_keymap('')` (i.e., empty filter) returns all items.
```
This only returns mappings in Normal, Visual, Select and operator-pending modes.

justinmk replied:
@zeertzjq isn't that a TODO, not an intentional design? For purposes of this example it's orthogonal, no?

justinmk commented on runtime/doc/lua.txt:
```diff
@@ -11,30 +11,126 @@ Lua engine                                                           *lua* *Lua*
+so that old plugins continue to work transparently.
+
+------------------------------------------------------------------------------
+LUA CONCEPTS AND IDIOMS                                         *lua-concepts*
```
Added `:help lua-concepts` to introduce key ideas and patterns of Lua. And moved the existing "syntax" section here.

justinmk commented on runtime/doc/lua.txt:
```diff
@@ -121,163 +217,6 @@ Note:
-through |vim.regex()|.
-
-------------------------------------------------------------------------------
-LUA PLUGIN EXAMPLE                                       *lua-require-example*
```
Removed this section entirely, it was not carrying its weight. Since Lua plugins work like Vimscript plugins, we can consider adding a small blurb about that somewhere instead.

clason commented on CONTRIBUTING.md:
```diff
@@ -47,21 +47,19 @@ Pull requests (PRs)
+- Use a _rebase workflow_ for small PRs.
+  - After addressing review comments, it's fine to force-push.
+- Use a _merge workflow_ (as opposed to "rebase") for big, high-risk PRs.
   - Merge `master` into your PR when there are conflicts or when master
     introduces breaking changes.
```
Are we sure we want to encourage people merging `master` into a PR, _especially_ a big one that is likely to get merged as opposed to squashed?

Since we now have a separate section on "Merging to master", we could remove these lines (or, better, replace them with a recommendation for contributors to "rebase on master instead of merging master into PR").

justinmk replied:
can revisit this later, but rebase causes different issues if there are lots of review comments already and many conflicting commits.

clason commented on runtime/doc/lua.txt:
```diff
@@ -11,30 +11,126 @@ Lua engine                                                           *lua* *Lua*
+
+The |:source| and |:runtime| commands can run Lua scripts. Lua modules can be
+loaded with `require('name')`, which by convention usually returns a table.
+See |lua-require| for how Nvim finds and loads Lua modules.
```
Would it be reasonable to add a (short!) note pointing out the difference between "script" and "module" here (or at least making clear that there _is_ a difference)?

justinmk replied:
what is the difference?

clason replied:
Modules are cached, for one thing.

justinmk replied:
That sounds like something worth calling out in `:help lua-require` or maybe `lua-concepts` ? Probably a bit low-level for the intro.

clason commented:
Hot take: It's always bugged me that the help file on scripting and plugins is called `repeat.txt`. Do we want to take this opportunity to change to a more sensible filename?

justinmk commented:
> Hot take: It's always bugged me that the help file on scripting and plugins is called `repeat.txt`. Do we want to take this opportunity to change to a more sensible filename?

out of scope for this PR (don't want to block it longer)

justinmk commented:
Followup incoming

justinmk commented:
/backport
